app/uk/gov/hmrc/tai/service/JourneyCacheService.scala

```diff
@@ -88,6 +88,21 @@ trait JourneyCacheService extends JourneyCacheConstants {
     }
   }
 
+  def collectedOptionalValues(optionalValues: Seq[String])(implicit hc: HeaderCarrier): Future[Seq[Option[String]]] = {
```

Comment:
Can you write unit tests for this method please?

Reply:
For consistency with the `mandatoryValues()` equivalent, I reckon we should also accept varargs param, and name it just 'optionalValues` - otherwise we've got an inconsistent interface on the JCS.
You could also call your new method from within collectedValues now, and remove second half of that method - since it's basically your method - would tidy your code (not essential though :)